web/src/server_events_dispatch.js

```diff
@@ -706,6 +706,18 @@ export function dispatch_normal_event(event) {
                 break;
             }
 
+            const privacy_settings = [
+                "send_stream_typing_notifications",
+                "send_private_typing_notifications",
+                "send_read_receipts",
+            ];
+
+            if (privacy_settings.includes(event.property)) {
+                user_settings[event.property] = event.value;
+                settings_account.update_privacy_settings_box(event.property);
+                break;
+            }
```

Comment:
I feel like the traditional pattern here would be to only call `settings_account` and put this after `user_preferences`, without pulling it out of that list. 

(This might a better pattern in that one can imagine `user_preferences` being generated from some other data source in the future).

Reply:
```
                "send_stream_typing_notifications",
                "send_private_typing_notifications",
                "send_read_receipts",
 ```
 
 These three are not present in "Preferences" tab. They are present in "Account & Privacy" tab, so I thought it makes sense to keep them in a separate list ?
 
 If we want to keep them in the user_preferences list, still I think a `privacy_settings` list (a subarray of the user_preferences array ) would be helpful for something like: 
 ```js
  if (privacy_settings.includes(event.property)) {
       settings_account.update_privacy_settings_box(event.property);
       if (event.property === "presence_enabled") {
           activity_ui.redraw_user(current_user.user_id);
       }
       break;
  }
```

Reply:
Hmm, reading a bit more, it does look like it's useful to have the distinctions you have. So I'll merge it as is, thanks.

@sahil839 FYI.